### PR TITLE
Add basic documentation for CRRL, CRAM, CSub, CMove, CClearTag

### PR DESCRIPTION
**Pull request by arichardson**

### CODE REVIEW

File: src/cheri_insts.sail [hunk 1]
```diff
@@ -309,6 +309,10 @@ function clause execute(CToPtr(rd, cs1, cs2)) =
 }
 
 union clause ast = CSub : (regidx, regidx, regidx)
+/*!
+ * Integer register *rd* is set equal to (*cs1*.**address** $-$
+ * *cs1*.**address**) $\bmod~2^{[{xlen}][xlen]}$.
```
Comment (PeterRugg) [inline src/cheri_insts.sail:314]:
```suggestion
 * *cs2*.**address**) $\bmod~2^{[{xlen}][xlen]}$.
```

File: src/cheri_insts.sail [hunk 2]
```diff
@@ -620,8 +631,12 @@ function clause execute (CCopyType(cd, cs1, cs2)) =
   }
 }
 
-
 union clause ast = CRRL : (regidx, regidx)
+/*!
+ * Integer register *rt* is set to the smallest value greater or equal to *rs1*
```
Comment (PeterRugg) [inline src/cheri_insts.sail:636]:
```suggestion
 * Integer register *rd* is set to the smallest value greater or equal to *rs1*
```

File: src/cheri_insts.sail [hunk 3]
```diff
@@ -620,8 +631,12 @@ function clause execute (CCopyType(cd, cs1, cs2)) =
...
+ * that can be used to set precise bounds on a capability that has a suitably
```
Comment (PeterRugg) [inline src/cheri_insts.sail:637]:
```suggestion
 * that can be used as a length to set precise bounds on a capability that has a suitably
```

File: src/cheri_insts.sail [hunk 4]
```diff
@@ -630,6 +645,11 @@ function clause execute(CRRL(rd, rs1)) =
 }
 
 union clause ast = CRAM : (regidx, regidx)
+/*!
+ * Integer register *rt* is set to a mask that can be used to round addresses
```
Comment (PeterRugg) [inline src/cheri_insts.sail:649]:
```suggestion
 * Integer register *rd* is set to a mask that can be used to round addresses
```

Comment (PeterRugg) [reply src/cheri_insts.sail:649]:
```suggestion
 * Integer register *rt* is set to the most permissive mask that can be used to round addresses
```
Assuming this is in fact true...

**Review by PeterRugg — CHANGES_REQUESTED** [review]

File: src/cheri_insts.sail (hunk 4 above)
Comment (arichardson) [reply src/cheri_insts.sail]: I am not sure this is strictly the most permissive mask. I think CRAM right now is precise?

Comment (PeterRugg) [reply src/cheri_insts.sail]: Hmm, I'm not sure: it's based on CSetBounds which should be precise. We should probably leave it. Just noticed the asymmetry with CRRL's "smallest value greater or equal".

**Review by PeterRugg — APPROVED** [review]

File: src/cheri_insts.sail (hunk 4 above)
Comment (arichardson) [reply src/cheri_insts.sail]: True, smallest in CRRL may also be an overstatement. But if I use "a value" there, then returning UINT64_MAX is a valid implementation.

Comment (PeterRugg) [reply src/cheri_insts.sail]: Just as returning 0 here is a valid implementation.

Comment (jrtc27) [reply src/cheri_insts.sail]: UINT64_MAX isn't representable, even if some existing capabilities claim to have that length :)

Comment (PeterRugg) [reply src/cheri_insts.sail]: Hmm, that's actually a good point. I recall conversations about this previously, but is CRRL specified to give UINT64_MAX in that case, or 0?

**Pull request merged**